### ISSUE TITLE
0.13.6 - java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError Regression

### ISSUE THREAD

**Issue by mdedetrich** [sbt/sbt#1610]
When upgrading to SBT 0.13.6, building our code makes SBT to run out of memory. Switching to SBT v0.13.5 immediately fixed the issue, so there is definitely a regression in this area.

This is private company code, so it may be hard to diagnose, but hopefully this stack trace can help

The code makes heavy use of implicits, so this may help in solving the regression (although I have no real idea if that is what caused the issue)

https://gist.github.com/mdedetrich/7dfa12e8c9e584b4d6a2

**Comment by gkossakowski** [comment]
Hi!

You can keep sbt 0.13.6 but turn off name hashing incremental compilation algorithm with `incOptions := incOptions.value.withNameHashing(false)`. How much memory does sbt process have? Does increasing memory limit help?

Could you take a memory dump and analyze what occupies most of the memory?

**Comment by mdedetrich** [comment]
The SBT process had the following parameters

```
SBT_OPTS="-XX:+UseConcMarkSweepGC -XX:+CMSClassUnloadingEnabled -Xms14g -Xmx14g"
```

This is on the latest and fully upgraded macbook pro. I tried different options (Xms/xmx used to be 8g) and it made no difference)

Regarding memory dump, do you have any instructions for this?

**Comment by gkossakowski** [comment]
Yes, take a look at:

http://stackoverflow.com/questions/542979/using-heapdumponoutofmemoryerror-parameter-for-heap-dump-for-jboss

One you have memory dump file created you can browse it using tools like YourKit or VisualVM. It appears like there's a big memory leak so it should be fairly easy to identify what's going on just by inspecting the heap dump.

**Comment by gkossakowski** [comment]
This has been fixed by #1642. See #1640.
